### DISCUSSION TITLE
extract frequent sequence

### DISCUSSION THREAD

**Discussion by deega4000** [UdayLab/PAMI#77] [General]
I am really impressed by this library. I think there is no other library that can recognise sequence.
I am trying to use it to find the frequent patterns in the actions such as stand ,sit ,go ,run.. etc 
The data looks as below: 
1 go,stand
2 stand,stand,chair,sleep,sit
3 go,go
4 stand,chair,sleep,sit
5 go,stand,go,stand
I tried FPGrowth algorithm but I think my data is not getting recognised properly. Also, my requirement is to get frequent sequence for ex. go,stand from the above example. If this works out, it will be great help.

**Comment by udayRage** [comment]
Dear Deepika,

Good day!

    Thank you very much for showing interest in our library.

    Coming to your query:

   1) In your sample dataset, an object can appear multiple times in a dataset (e.g., stand, stand in row-2). This dataset is not a transactional database. It is because an item appears only once in a transaction in a transactional database.

   2) Henceforth, you cannot employ frequent pattern mining techniques, such as FP-growth.

  3) Your sample dataset resembles the sequence database. Please convert your dataset as follows
     go -1 stand -1 -2
     stand -1 stand -1 chair -1 sleep -1 sit -1 -2
    go -1 go -1 -2
    ...

   In the future version, -1 -2 at the line's end is unnecessary. 

  4) Implement the SPADE algorithm to discover frequent sequence patterns.

I have requested my student to prepare the manual for using the SPADE algorithm. He will share it with you in a few hours.
For the time being, please check the manual: https://github.com/UdayLab/PAMI/blob/main/notebooks/sequencePatternMining/basic/SPADE.ipynb

**Reply by deega4000** [reply]
Thanks for the reply. 
Yes you got it right, I have a sequence database and the actual data is quite complex where I have long sequences of around 20 steps so I am not sure if I can create the required format quickly.
Can I find the sample dataset anywhere or the logic behind the below 
go -1 stand -1 -2
stand -1 stand -1 chair -1 sleep -1 sit -1 -2
go -1 go -1 -2

**Comment by udayRage** [comment]
Please check the manual shared above to find the sample dataset.
        URL: https://github.com/UdayLab/PAMI/blob/main/notebooks/sequencePatternMining/basic/SPADE.ipynb

A sequence database is a sets of itemsets.
E.g.,

 {{Bread, Jam}, {Beer, Diapers, milk},{Bat, Ball}}
{{Bread, Bat, Ball}, {Milk}}
....

We store the above sample dataset in the below provided format for ease of parsing.

 Bread Jam -1 Beer Diapers milk -1 bat ball -1 -2.   
 Bread Bat Ball -1 Milk -1 -2
 ...

  -1 represents the end of a set of items
  -2 represents the end of the sequence (or sets of itemsets).
-----------------
I have long sequences of around 20 steps so I am not sure if I can create the required format quickly.
Ans) Unfortunately, I can't help here as it is beyond my scope.
